ls1intum/Artemis
# Improvements to BambooService

Various improvements in BambooService

### Checklist

### Motivation and Context
We would like to get rid of the used external library and implement the REST requests on our own.
This allows for more flexibility and means less dependencies on other libraries

### Description
* Re-implement a couple of REST calls between Artemis and Bamboo
* Made token authentication in BambooService mandatory
* Change `secret` to `token` for Jenkins
* No client changes

### Steps for Testing on TS1 and TS2

1. Create a programming exercise A
2. Participate in programming exercise A and check that everything works properly
3. Import programming exercise A into programming exercise B
4. Participate in programming exercise B and check that everything works properly
5. Delete the programming exercise A and B

### Test Coverage
I improved the test coverage for BambooService (now > 80%) and for BitbucketService (now 51%)
